Makefile

```diff
@@ -13,7 +13,7 @@ occ=$(CURDIR)/../../occ
 
 # bin file definitions
 PHPUNIT=php -d zend.enable_gc=0  ../../lib/composer/bin/phpunit
-PHPUNITDBG=phpdbg -qrr -d memory_limit=4096M -d zend.enable_gc=0 "../../lib/composer/bin/phpunit"
+PHPUNITDBG=phpdbg -qrr -d memory_limit=8192M -d zend.enable_gc=0 "../../lib/composer/bin/phpunit"
```

Comment:
@patrickjahns are we ok with this or will Drone explode eventually ?

Reply:
@sharidas I do wonder what is using all that memory. Could it be that our JSON has become too big ? But even then it shouldn't take Gigabytes

Reply:
The constructor has https://github.com/owncloud/configreport/blob/master/lib/ReportDataCollector.php#L132 which is causing the problem :( I am debugging further to find the exact cause.

Reply:
Machine frozen twice while debugging `listAllApps()` :(

Reply:
Please don't raise the limit to 8GB - 4GB is more than enough.
If you need to raise the memory limit - we most probably deal with a memory leak somewhere ( recursion? ).

Reply:
> Please don't raise the limit to 8GB - 4GB is more than enough.
> If you need to raise the memory limit - we most probably deal with a memory leak somewhere ( recursion? ).

Its happening in the loop https://github.com/owncloud/core/blob/master/lib/private/legacy/app.php#L781. While getting all the apps, the allocated memory 4GB is getting exhausted. For example  below is the number of bytes used in the loop ( captured with `memory_get_usage()` )
```
29001488
882905720
1734686240
1734757560
1734824432
1734824432
1734961288
2586471872
3436802864
3436802864
```

Reply:
maybe there's an infinite loop ? does the function that gather the apps also run the app's "app.php" implicitly ? are you able to get a stack trace or debug into it ?

Reply:
The my investigation is making me believe that the problem is with the line https://github.com/owncloud/core/blob/master/lib/private/legacy/app.php#L812. I got the trace The trace is little large. I am using xdebug trace to grab the output to a file. The file size is little large ( ~2GB) . So need to trim it down. But yes the signature verification of the app is giving lot of function call in the trace. 

Why do I believe the problem is with the line https://github.com/owncloud/core/blob/master/lib/private/legacy/app.php#L812?
- When the line is commented and added `$result = '';` ( to confirm that I am in the right direction ), the phpunit is executed in fast manner.

Reply:
Drilled down to two methods which are the memory eaters :(
- https://github.com/owncloud/core/blob/master/lib/private/IntegrityCheck/Checker.php#L342
- https://github.com/owncloud/core/blob/master/lib/private/IntegrityCheck/Checker.php#L352

Both validate signature methods are causing trouble. Debugging further.

Reply:
As per my research the problem is here https://github.com/phpseclib/phpseclib/blob/2.0/phpseclib/Crypt/RSA.php#L2821-L2835

Reply:
why would that routine produce a huge stack ?

Reply:
> I wonder if there are also any symlinks involved as it would likely make the integrity checker run in circles

I did not found any symlinks in the apps folder. Traversed through each apps and verified.

Reply:
> As per my research the problem is here https://github.com/phpseclib/phpseclib/blob/2.0/phpseclib/Crypt/RSA.php#L2821-L2835

Again drilled down to https://github.com/phpseclib/phpseclib/blob/2.0/phpseclib/Math/BigInteger.php#L599-L602

### Remark ###
This issue is caused not when any comparison is made with the hash value in `appinfo/signature.json`. This issue is caused when the code tries to verify signatures in file
- `resources/codesigning/root.crt`
- `resources/codesigning/intermediate.crl.pem`

The hashes in the signature.json are done at the last phase of `verify()` method in the Checker.php.

Reply:
down the rabbit hole :laughing: